terraform/product/mongodb-sharded-cluster/variables.tf

```
@@ -0,0 +1,148 @@
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+    expose            = optional(bool, false)
+  })
+}
+
+variable "apps" {
+  description = "Non orchestrator apps (e.g: ml, data.hot etc.)"
+  type = list(object({
+    app_name          = string
+    model             = string
+    config            = optional(map(string), {role="shard"})
+    channel           = optional(string, "6/stable")
+    base              = optional(string, "ubuntu@22.04")
+    revision          = optional(string, 199)
+    units             = optional(number, 3)
+    constraints       = optional(string, "arch=amd64")
+    machines          = optional(list(string), [])
+    storage           = optional(map(string), {})
+    endpoint_bindings = optional(map(string), {})
+    expose            = optional(bool, false)
+  }))
+  default = null
+}
+
+# --------
+variable "self-signed-certificates" {
+  description = "Configuration for the self-signed-certificates app"
+  type = object({
+    channel     = optional(string, "latest/stable")
+    revision    = optional(string, null)
+    base        = optional(string, "ubuntu@22.04")
+    constraints = optional(string, "arch=amd64")
+    machines    = optional(list(string), [])
+    config      = optional(map(string), { "ca-common-name" : "CA" })
+  })
+  default = {}
+
+  validation {
+    condition     = length(var.self-signed-certificates.machines) <= 1
+    error_message = "Machine count should be at most 1"
+  }
+}
+
+variable "grafana-agent" {
+  description = "Configuration for the grafana-agent"
+  type = object({
+    channel     = optional(string, "1/stable")
+    revision    = optional(string, null)
+    base        = optional(string, "ubuntu@22.04")
+    constraints = optional(string, "arch=amd64")
+    machines    = optional(list(string), [])
+    config      = optional(map(string), {})
+  })
+  default = {}
+}
+
+# Integrators
+variable "backups-integrator" {
+  description = "Configuration for the backup integrator"
+  type = object({
+    storage_type = optional(string, "s3")
+    config       = map(string)
+    channel      = optional(string, "latest/stable")
+    base         = optional(string, "ubuntu@22.04")
+    revision     = optional(string, null)
+    constraints  = optional(string, "arch=amd64")
+    machines     = optional(list(string), [])
+  })
+
+  validation {
+    condition     = contains(["s3", "azure-storage"], var.backups-integrator.storage_type)
+    error_message = "backup-integrator allows one of the values: 's3', 'azure' for storage_type."
+  }
+}
+
+variable "data-integrator" {
+  description = "Configuration for the data-integrator"
+  type = object({
+    config      = optional(map(string), { "index-name" : "test", "extra-user-roles" : "admin" })
```

On mongodb charms we're not providing indexes but databases.
The configuration option is named `database-name`.
```suggestion
    config      = optional(map(string), { "database-name" : "test", "extra-user-roles" : "admin" })
```